src/codecs/encoding/transformer.rs

```diff
@@ -141,7 +153,20 @@ impl Transformer {
     fn apply_except_fields(&self, log: &mut LogEvent) {
         if let Some(except_fields) = self.except_fields.as_ref() {
             for field in except_fields {
-                log.remove(field.as_str());
+                let value = log.remove(field.as_str());
+
+                // If we are removing the service field we need to store this in a `dropped_fields` list as we may need to
+                // refer to this later when emitting metrics.
+                if let Some(v) = value {
+                    if let Some(path) = log.metadata().schema_definition().meaning_path("service") {
+                        use lookup::path::TargetPath;
```

Comment:
I think it's unusual seeing `use` statements this deep in the code.

Reply:
Hmm.. I'm not sure if I want that trait to be available in the whole module because it adds function to `&str`. So I am trying to limit the scope to just here.

Reply:
I've reorganised the function a bit.

Reply:
Got it, just making sure it was intentional - I've had IDE's sometimes stick those in more willy-nilly than I'd like.